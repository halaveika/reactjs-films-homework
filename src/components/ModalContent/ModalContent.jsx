import React from 'react';
import PropTypes from 'prop-types';
import './ModalContent.scss';

export default function ModalContent({ video, videoRef }) {
  return (
    (video)

      ? (
        <iframe
          src={`${video}`}
          scrolling="no"
          frameBorder="0"
          allow="accelerometer; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen
          ref={videoRef}
          title="trailer"
        />
      )

      : <img className="image-notfound" ref={videoRef} src="assets/images/not-found.png" alt="notfound" />
  );
}

ModalContent.propTypes = {
  video: PropTypes.string.isRequired,
  videoRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: null }),
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
};
