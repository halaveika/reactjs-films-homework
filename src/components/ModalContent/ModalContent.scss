.ant-modal {
  top: 0;
}

.ant-modal-content {
  width: 100vw;
  height: 100vh;
}

.ant-modal-body {
  width: 100%;
  height: 90%;
}

.ant-modal-footer {
  width: 100%;
  height: 10%;
}

iframe {
  margin: 0 auto;
  width: 100%;
  height: 100%;
}

.image-notfound {
  display: block;
  margin: 0 auto;
}
