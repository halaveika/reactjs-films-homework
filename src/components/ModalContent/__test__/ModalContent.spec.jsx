import React from 'react';
import { create } from 'react-test-renderer';
import ModalContent from '../ModalContent';

const videoRef = { current: null };

describe('test ModalContent component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render ModalContent as iframe', () => {
    const component = create(
      <ModalContent
        video="test"
        videoRef={videoRef}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sshould render ModalContent as image', () => {
    const component = create(
      <ModalContent
        video=""
        videoRef={videoRef}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
