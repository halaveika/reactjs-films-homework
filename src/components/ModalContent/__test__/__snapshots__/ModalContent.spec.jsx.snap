// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ModalContent component should render ModalContent as iframe 1`] = `
<iframe
  allow="accelerometer; encrypted-media; gyroscope; picture-in-picture"
  allowFullScreen={true}
  frameBorder="0"
  scrolling="no"
  src="test"
  title="trailer"
/>
`;

exports[`test ModalContent component sshould render ModalContent as image 1`] = `
<img
  alt="notfound"
  className="image-notfound"
  src="assets/images/not-found.png"
/>
`;
